allennlp/tests/modules/token_embedders/pretrained_transformer_pretokenized_embedder_test.py

```diff
@@ -0,0 +1,54 @@
+import torch
+
+from allennlp.common import Params
+from allennlp.data import Token, Vocabulary
+from allennlp.data.batch import Batch
+from allennlp.data.fields import TextField
+from allennlp.data.instance import Instance
+from allennlp.data.token_indexers import PretrainedTransformerPretokenizedIndexer
+from allennlp.modules.text_field_embedders import BasicTextFieldEmbedder
+from allennlp.common.testing import AllenNlpTestCase
+
+
+class TestPretrainedTransformerEmbedder(AllenNlpTestCase):
+    def test_end_to_end(self):
+        token_indexer = PretrainedTransformerPretokenizedIndexer("bert-base-uncased")
+
+        sentence1 = ["A", ",", "AllenNLP", "sentence", "."]
+        sentence2 = ["AllenNLP", "is", "great"]
+        tokens1 = [Token(word) for word in sentence1]
+        tokens2 = [Token(word) for word in sentence2]
+
+        vocab = Vocabulary()
+
+        params = Params(
+            {
+                "token_embedders": {
+                    "bert": {
+                        "type": "pretrained_transformer_pretokenized",
+                        "model_name": "bert-base-uncased",
+                    }
+                }
+            }
+        )
+        token_embedder = BasicTextFieldEmbedder.from_params(vocab=vocab, params=params)
+
+        instance1 = Instance({"tokens": TextField(tokens1, {"bert": token_indexer})})
+        instance2 = Instance({"tokens": TextField(tokens2, {"bert": token_indexer})})
+
+        batch = Batch([instance1, instance2])
+        batch.index_instances(vocab)
+
+        padding_lengths = batch.get_padding_lengths()
+        tensor_dict = batch.as_tensor_dict(padding_lengths)
+        tokens = tensor_dict["tokens"]
+
+        assert tokens["bert"]["offsets"].tolist() == [
```

Comment:
Oh, you're testing it here.  This assert should live in the indexer test, shouldn't it?

Reply:
Can you elaborate on this and the coverage issue you mentioned above? I do check the `offsets` key in the indexer test. I'm just checking here again, just like `TestPretrainedTransformerEmbedder.test_end_to_end` also tests some indexer input, e.g. it has `assert tokens["bert"]["mask"].tolist() == X`.

Reply:
The difference between this test and the last one is that this is checking after padding has been added.  That's particularly the line that you're relying on the superclass for.  I'd prefer a test that checks for padding on all keys that are output by this class.